docs/source/reference/samplers.rst

```diff
@@ -20,5 +20,12 @@ The :mod:`~optuna.samplers` module defines a base class for parameter sampling a
    optuna.samplers.PartialFixedSampler
    optuna.samplers.NSGAIISampler
    optuna.samplers.MOTPESampler
+   optuna.samplers.BaseCrossover
+   optuna.samplers.UniformCrossover
+   optuna.samplers.BLXAlphaCrossover
+   optuna.samplers.SPXCrossover
+   optuna.samplers.SBXCrossover
+   optuna.samplers.VSBXCrossover
+   optuna.samplers.UNDXCrossover
```

Comment:
It's great that you are rising this, since I'm not a big fan of current structure myself (but wanted to keep it simple in the first proposition). `optuna.samplers.nsga2.BaseCrossover` sounds much more reasonable, but I'd like to hear your opinion on sampler itself - are we keeping it as is or also make it part of `nsga2` import structure? (my bet is to keep it as is.)

Reply:
I think it is better to keep as it is. Changing the import structure of `NSGAIISampler` should not be done because of its high impact on users.

Reply:
Great, I'll move only crossovers then.

Reply:
@HideakiImamura

I pushed few commits implementing the changres. Crossovers are now separated from samplers and reside in NSGAII specific directory. We can now import them like this

![Screenshot from 2022-01-26 21-20-25](https://user-images.githubusercontent.com/37713008/151241324-e0a35564-060f-4c2d-b03a-1a72b963fcd9.png)

or like that

![Screenshot from 2022-01-26 21-25-53](https://user-images.githubusercontent.com/37713008/151241470-f8d072a7-e630-4b5f-bf98-efbb683f01ed.png)

API reference for samplers now has NSGAII specific subtree as well.

![Screenshot from 2022-01-26 21-27-50](https://user-images.githubusercontent.com/37713008/151241750-68776c6b-0c54-4ad0-a6e5-0cce2edf78b5.png)